src/shared/components/lock-screen/LockScreen.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import Lottie from 'lottie-react-web';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import styles from './LockScreen.css';
+import animation from '../../media/animations/unlock-animation.json';
+import background from '../../media/images/background-pattern.svg';
+
+const LOCK_TYPE = 'passphrase';
+
+const lockscreenContentClassNames = {
+    appear: styles.lockscreenContentAppear,
+    appearActive: styles.lockscreenContentAppearActive,
+    exit: styles.lockscreenContentExit,
+    exitActive: styles.lockscreenContentExitActive,
+    exitDone: styles.lockscreenContentExitDone,
+};
+
+const passwordDotsClassNames = {
+    enter: styles.passwordDotEnter,
+    enterActive: styles.passwordDotEnterActive,
+    exit: styles.passwordDotExit,
+    exitActive: styles.passwordDotExitActive,
+};
+
+class LockScreen extends Component {
+    passwordInputRef = React.createRef();
+    animationOptions = {
+        animationData: animation,
+        loop: false,
+    };
+
+    state = {
+        passwordLength: 0,
+        feedback: 'none',
+        startLogoAnimation: false,
+    };
+
+    componentDidMount() {
+        const input = this.passwordInputRef.current;
+
+        input.focus();
+        input.addEventListener('keydown', this.handleKeyboardInput);
+        input.addEventListener('blur', () => input.focus());
+
+        this.mounted = false;
+        this.forceUpdate();
+
+        console.log('styles.lockscreenContentAppearActive', styles.lockscreenContentAppearActive);
+    }
+
+    render() {
+        const { feedback, startLogoAnimation } = this.state;
+        const { unmounting } = this.props;
+
+        return (
+            <div className={ styles.lockscreen } onClick={ this.handleMouseClick }>
+                <div className={ styles.background } dangerouslySetInnerHTML={ { __html: background } } />
+                <CSSTransition classNames={ lockscreenContentClassNames } in={ !unmounting } appear component={ null } timeout={ 1500 }>
+                    <div className={ styles.lockscreenContent }>
+                        <div className={ styles.logo }>
+                            <Lottie options={ this.animationOptions } className={ styles.svg } isPaused={ !startLogoAnimation } />
+                        </div>
+                        <h2 className={ styles.unlockTitle }>Unlock Nomios</h2>
+                        <p className={ styles.unlockHint } onTransitionEnd={ this.handleTransitionEnd }>
+                            Enter your passphrase to unlock Nomios and get access to all your data
+                        </p>
+                        <input type="password" name="getLockKey" className={ styles.passwordInput }
+                            ref={ this.passwordInputRef } onChange={ this.handlePasswordChange } disabled={ feedback === 'loading' } />
```

Comment:
I would break this props in one by one. When it gets to big, one by one becomes a lot easier to read.